ecs_fargate/README.md

```diff
@@ -8,65 +8,92 @@ Get metrics from all your containers running in ECS Fargate:
-
-### Deploying the Datadog Agent on AWS Fargate
+The following steps cover setup of the Datadog Container Agent within AWS ECS Fargate. **Note**: Datadog Agent version 6.1.1 or higher is needed to take full advantage of the Fargate integration.
 
-To start monitoring AWS Fargate tasks and services, deploy the containerized Datadog Agent on Fargate. In addition to passing the usual `DD_API_KEY` environment variable, you must set the `ECS_FARGATE `environment variable to `true`.
+### Installation
+To monitor your ECS Fargate containers with Datadog, run the Agent as a container in same task definition as your application. To collect metrics with Datadog, each task definition requires a Datadog Agent container. Follow these setup steps:
 
-The example task definition below deploys the Datadog Agent to Fargate, along with a Redis container in the same task.
+1. **Add an ECS Fargate Task**
+2. **Create or Modify your IAM Policy**
+3. **Run the Datadog Agent as a Replica Service**
 
-```
-{'containerDefinitions': [{'dockerLabels': {'com.datadoghq.ad.check_names': '["redisdb"]',
-                                            'com.datadoghq.ad.init_configs': '[{}]',
-                                            'com.datadoghq.ad.instances': '[{"host": "%%host%%", "port": 6379}]'},
-                           'essential': true,
-                           'image': 'redis:latest',
-                           'name': 'redis'},
-                          {'environment': [{'name': 'DD_API_KEY',
-                                            'value': '$YOUR_API_KEY'},
-                                           {'name': 'ECS_FARGATE',
-                                            'value': 'true'}],
-                           'essential': true,
-                           'image': 'datadog/agent:latest',
-                           'name': 'datadog-agent'}],
- 'cpu': '256',
- 'family': 'redis-datadog',
- 'memory': '512',
- 'networkMode': 'awsvpc',
- 'requiresCompatibilities': ['FARGATE']}
-```
+#### Create an ECS Fargate Task
+This task launches the Datadog Agent container. Configure the task using the [AWS CLI tools][11] or the [Amazon Web Console][12].
 
-After saving that task definition locally as `redis-datadog.json`, register the task with ECS using the [AWS CLI][11]:
+##### AWS CLI
 
+1. Download [datadog-agent-ecs-fargate.json][18].
+2. Update the json with a **TASK_NAME** and your [Datadog API Key][13].
+3. Execute the following command to register the ECS task definition:
 ```
-aws ecs register-task-definition --cli-input-json file://./redis-datadog.json
+aws ecs register-task-definition --cli-input-json file://<PATH_TO_FILE>/datadog-agent-ecs-fargate.json
 ```
 
-Once the task has been registered, you can run it in a Fargate cluster, making sure to provide one or more private subnets and security groups for your task. Note that Fargate version 1.1.0 or greater is required, so specify the platform version in the command:
+##### Web UI
```

Comment:
same here, these instructions are good, but they need to include the agent in an app task.

For example step `4.` is misleading. We don't want a `datadog-agent-task`, we want a `my-app-and-datadog` task that already exists, and where users follow step 7 through 14 to add the agent alongside their app.